static/js/message_list_data.js

```diff
@@ -182,12 +183,34 @@ export class MessageListData {
         });
     }
 
+    messages_filtered_for_user_mutes(messages) {
+        if (!this.excludes_muted_users) {
+            return [...messages];
+        }
+
+        return messages.filter((message) => {
+            if (message.type !== "private") {
+                return true;
+            }
+            const recipients = util.extract_pm_recipients(message.to_user_ids);
+            if (recipients.length > 1) {
+                // Huddle message
+                return true;
+            }
+
+            const recipient_id = Number.parseInt(recipients[0], 10);
+            return !muting.is_user_muted(recipient_id) && !muting.is_user_muted(message.sender_id);
+        });
+    }
```

Comment:
I think it'd be cleaner to just do this in the same function that does topic mutes, rather than having two predicates, since it appears we're just going to compose these functions when they're called anyway.

Reply:
> since it appears we're just going to compose these functions when they're called anyway.

Actually, we can we can further use the fact that these are independent in the wrapper methods in `message_list.js`.
So, both of these methods will be composed only when adding messages etc to the MLD object, and only one of these will be run when we want to update the message list after receiving events.

I pushed this version here. It doesn't save on complexity/performance, but I think reads a bit better. Does that seem reasonable?

Reply:
I think that makes sense to me.

Reply:
See https://github.com/zulip/zulip/pull/16915/commits/28e14ffe4c799ff0e537c87cb966f7acaee17b0e#r627861439 but I think in retrospect this logic doesn't actually work.